x/protocolpool/keeper/keeper.go

```diff
@@ -67,3 +88,115 @@ func (k Keeper) GetCommunityPool(ctx context.Context) (sdk.Coins, error) {
 	}
 	return k.bankKeeper.GetAllBalances(ctx, moduleAccount.GetAddress()), nil
 }
+
+func (k Keeper) claimFunds(ctx context.Context, recipient sdk.AccAddress) (amount sdk.Coin, err error) {
+	// get claimable funds from distribution info
+	amount, err = k.getClaimableFunds(ctx, recipient)
+	if err != nil {
+		return sdk.Coin{}, err
+	}
+
+	// distribute amount from feepool
+	err = k.DistributeFromFeePool(ctx, sdk.NewCoins(amount), recipient)
+	if err != nil {
+		return sdk.Coin{}, err
+	}
+
+	return amount, nil
+}
+
+func (k Keeper) getClaimableFunds(ctx context.Context, recipient sdk.AccAddress) (amount sdk.Coin, err error) {
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+
+	budget, err := k.BudgetProposal.Get(ctx, recipient)
+	if err != nil {
+		if errors.Is(err, collections.ErrNotFound) {
+			return sdk.Coin{}, fmt.Errorf("no budget found for recipient: %s", recipient.String())
+		}
+		return sdk.Coin{}, err
+	}
+
+	// check if the distribution is completed
+	if budget.Tranches <= 0 {
+		// remove the entry of budget ended recipient
+		err := k.BudgetProposal.Remove(ctx, recipient)
```

Comment:
if the tranches are zero but the claimed amount is greater than 1 we probably shouldnt remove, right?